core/src/main/java/io/confluent/rest/metrics/MetricsResourceMethodApplicationListener.java

```diff
@@ -184,57 +183,65 @@ private static class MethodMetrics {
     private final Sensor[] errorSensorByStatus = new Sensor[HTTP_STATUS_CODE_TEXT.length];
 
     public MethodMetrics(ResourceMethod method, PerformanceMetric annotation, Metrics metrics,
-                         String metricGrpPrefix, Map<String, String> metricTags) {
+                         String metricGrpPrefix, Map<String, String> metricTags,
+                         Map<String, String> requestTags) {
       String metricGrpName = metricGrpPrefix + "-metrics";
+      // The tags will be used to generate MBean names if JmxReporter is used,
+      // sort to get consistent names
+      Map<String, String> allTags = new TreeMap<>(metricTags);
+      allTags.putAll(requestTags);
 
-      this.requestSizeSensor = metrics.sensor(getName(method, annotation, "request-size"));
+      this.requestSizeSensor = metrics.sensor(
```

Comment:
When making a similar change in kafka-rest, I was asked to ensure the sensors expire if they are unused, otherwise we will end up with a sensor leak.

1 hour was the suggested duration.

This didn't matter so much when one sensor was reused, but now there are multiple sensors I think it is important.

Reply:
Good point, addressed, thanks!